### PR TITLE
Fix for lun disks

### LINKED ISSUE #6507: Operation not permitted in setting unpriv_sgio  with scsi lun
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
If you want to use lun disk with scsi, the VM fails to start because libvirt cannot set unpriv_sgio

**How to reproduce it (as minimally and precisely as possible)**:
Start a VM with a block PVC:
```yaml
---
apiVersion: kubevirt.io/v1
kind: VirtualMachineInstance
metadata:
  labels:
    special: vmi-lun
  name: vmi-lun
spec:
  domain:
    devices:
      disks:
      - disk:
          bus: virtio
        name: containerdisk
      - lun:
          bus: scsi
        name: blockpvcdisk
    resources:
      requests:
        memory: 128Mi
  terminationGracePeriodSeconds: 0
  volumes:
  - containerDisk:
      image: registry:5000/kubevirt/cirros-container-disk-demo:devel
    name: containerdisk
  - name: blockpvcdisk
    persistentVolumeClaim:
      claimName: pvc-test
```

Check the virt-launcher log:
```bash
$ k logs -c  compute virt-launcher-vmi-lun-dhwg5
[...]
{"component":"virt-launcher","level":"error","msg":"failed to set /sys/dev/block/251:0/queue/unpriv_sgio: Operation not permitted","pos":"virSetDeviceUnprivSGIO:1417","subcomponent":"libvirt","thread":"31","timestamp":"2021-10-04T11:00:43.570000Z"}
{"component":"virt-launcher","kind":"","level":"error","msg":"Failed to start VirtualMachineInstance with flags 0.","name":"vmi-lun","namespace":"default","pos":"manager.go:814","reason":"virError(Code=38, Domain=0, Message='failed to set /sys/dev/block/251:0/queue/unpriv_sgio: Operation not permitted')","timestamp":"2021-10-04T11:00:43.591709Z","uid":"b3715891-d9a6-4c47-ba98-5d28368dbea3"}
{"component":"virt-launcher","kind":"","level":"error","msg":"Failed to sync vmi","name":"vmi-lun","namespace":"default","pos":"server.go:184","reason":"virError(Code=38, Domain=0, Message='failed to set /sys/dev/block/251:0/queue/unpriv_sgio: Operation not permitted')","timestamp":"2021-10-04T11:00:43.592120Z","uid":"b3715891-d9a6-4c47-ba98-5d28368dbea3"}
[...]
```
Xml:
```xml
    <disk type='block' device='lun'>
      <driver name='qemu' type='raw' cache='none' error_policy='stop' io='native' discard='unmap'/>
      <source dev='/dev/blockpvcdisk'/>
      <target dev='sda' bus='scsi'/>
      <alias name='ua-blockpvcdisk'/>
      <address type='drive' controller='0' bus='0' target='0' unit='0'/>
    </disk>

```

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2010306

### PR DESCRIPTION
**Pull request by alicefr**
Add functional test to check if a VMI can be started with lun disks.

Signed-off-by: Alice Frosi <afrosi@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes  #6507

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by alicefr** [comment]
The new libvirt version with the fix has been released and I bumped the version. Now the new functional tests should pass

fyi @andreabolognani

**Comment by alicefr** [comment]
I broke the `faulty disk` test need to take a look

**Comment by rmohr** [comment]
/approve

@mhenriks can you restore your lgtm if it is still fine?

File: tests/io_utils.go [hunk 1]
```diff
@@ -69,13 +69,19 @@ func ExecuteCommandInVirtHandlerPod(nodeName string, args []string) (stdout stri
 
 func CreateErrorDisk(nodeName string) (address string, device string) {
 	By("Creating error disk")
-	args := []string{"/usr/bin/virt-chroot", "--mount", "/proc/1/ns/mnt", "exec", "--", "/usr/sbin/modprobe", "scsi_debug", "opts=2", "every_nth=4"}
+	return CreateSCSIDisk(nodeName, []string{"opts=2", "every_nth=4"})
+}
+
+// CreateSCSIDisk creates a SCSI disk using the scsi_debug module. This function should be used only to check SCSI disk functionalities and not for creating a filesystem or any data. The disk is stored in ram and it isn't suitable for storing large amount of data.
+func CreateSCSIDisk(nodeName string, opts []string) (address string, device string) {
+	args := []string{"/usr/bin/virt-chroot", "--mount", "/proc/1/ns/mnt", "exec", "--", "/usr/sbin/modprobe", "scsi_debug"}
```
Comment (rmohr) [inline tests/io_utils.go:77]: `scsi_debug` sounds like something which we could enable in the kubevirtci environments.

Comment (rmohr) [inline tests/io_utils.go:77]: `scsi_debug` sounds like something which we could enable in the kubevirtci environments.

**Review by rmohr — COMMENTED** [review]
/lgtm

Restoring the lgtm from @ShellyKa13

File: tests/io_utils.go (hunk 1 above)
Comment (alicefr) [reply tests/io_utils.go:77]: @rmohr yes this might be a good idea.  I just move it from the IO error disk generation

**Pull request merged**